matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/aggregation/livelocation/DefaultLiveLocationAggregationProcessor.kt

```diff
@@ -17,69 +17,74 @@
-            return
-        }
-        val beaconInfoContent = ContentMapper.map(beaconInfoEntity.root?.content)?.toModel<LiveLocationBeaconContent>(catchError = true)
-        if (beaconInfoContent == null) {
-            Timber.v("## LIVE LOCATION. Beacon info content is invalid")
+        if (targetEventId.isNullOrEmpty()) {
+            Timber.w("no target event id found for the beacon content")
             return
         }
 
-        // Check if live location is ended
-        if (!beaconInfoContent.isLive.orFalse()) {
-            Timber.v("## LIVE LOCATION. Beacon info is not live anymore")
+        val aggregatedSummary = LiveLocationAggregatedSummaryEntity.getOrCreate(
+                realm = realm,
+                roomId = roomId,
+                eventId = targetEventId
+        )
+
+        aggregatedSummary.endOfLiveTimestampAsMilliseconds = content.getBestTimestampAsMilliseconds()?.let { it + (content.timeout ?: 0) }
+        aggregatedSummary.isLive = content.isLive
+    }
+
+    override fun handleLiveLocation(realm: Realm, event: Event, content: MessageLiveLocationContent, roomId: String, isLocalEcho: Boolean) {
+        if (event.senderId.isNullOrEmpty() || isLocalEcho) {
             return
         }
 
-        // Check if beacon info is outdated
-        if (isBeaconInfoOutdated(beaconInfoContent, content)) {
-            Timber.v("## LIVE LOCATION. Beacon info has timeout")
-            beaconInfoContent.hasTimedOut = true
-        } else {
-            beaconInfoContent.lastLocationContent = content
+        val targetEventId = content.relatesTo?.eventId
+
+        if (targetEventId.isNullOrEmpty()) {
+            Timber.w("no target event id found for the live location content")
+            return
         }
 
-        beaconInfoEntity.root?.content = ContentMapper.map(beaconInfoContent.toContent())
-    }
+        val aggregatedSummary = LiveLocationAggregatedSummaryEntity.getOrCreate(
+                realm = realm,
+                roomId = roomId,
+                eventId = targetEventId
+        )
+        val updatedLocationTimestamp = content.getBestTimestampAsMilliseconds() ?: 0
+        val currentLocationTimestamp = ContentMapper
+                .map(aggregatedSummary.lastLocationContent)
+                .toModel<MessageLiveLocationContent>()
+                ?.getBestTimestampAsMilliseconds()
+                ?: 0
 
-    private fun isBeaconInfoOutdated(beaconInfoContent: LiveLocationBeaconContent,
-                                     liveLocationContent: MessageLiveLocationContent): Boolean {
-        val beaconInfoStartTime = beaconInfoContent.getBestTimestampAsMilliseconds() ?: 0
-        val liveLocationEventTime = liveLocationContent.getBestTimestampAsMilliseconds() ?: 0
-        val timeout = beaconInfoContent.timeout ?: 0
-        return liveLocationEventTime - beaconInfoStartTime > timeout
+        if (updatedLocationTimestamp > currentLocationTimestamp) {
```

Comment:
we could extract a helper to avoid the comment

```kotlin
updatedLocationTimestamp.isMoreRecentThan(currentLocationTimestamp) {
  aggregatedSummary.lastLocationContent = ContentMapper.map(content.toContent())
}
```